### ISSUE TITLE
[Bug]: DatePicker Accessibility Issue aria-allowed-role

### ISSUE THREAD

**Issue by jwace81** [microsoft/fluentui#27781]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
Package version: @fluentui/react: 8.109.2
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker

## Repro Steps
- Navigate to the documentation for the DatePicker component: https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker
- Open a DatePicker so that the calendar is visible
- Run the Accessibility Insights for Web fast pass.

## Actual Behavior
The FluentUI DatePicker component is currently failing the Accessibility Insights for Web fast pass. See screenshots and attached logs.
![image](https://user-images.githubusercontent.com/29102072/236577969-17e401fe-581b-47a3-883b-06bfc576713e.png)
![image](https://user-images.githubusercontent.com/29102072/236578027-6f9325fc-ceef-49b2-a139-7b6653c9cf1c.png)

## Expected Behavior
The DatePicker component should pass the Accessibility Insights for Web fast pass without any accessibility issues.

### Logs

```shell
Title: ,WCAG 1.3.1,WCAG 4.1.2,Best Practice: Ensures role attribute has an appropriate value for the element (button[aria-label="January"])
Tags: Accessibility, , WCAG 1.3.1, WCAG 4.1.2, Best Practice, aria-allowed-role

Issue: Ensures role attribute has an appropriate value for the element (aria-allowed-role - https://dequeuniversity.com/rules/axe/4.6/aria-allowed-role?application=msftAI)

Target application: Fluent UI - Controls - React - DatePicker - https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker

Element path: button[aria-label="January"]

Snippet: <button role="gridcell" class="itemButton-497" aria-label="January" aria-selected="false" data-is-focusable="true" type="button" tabindex="0">Jan</button>

How to fix: 
Fix any of the following:
  ARIA role gridcell is not allowed for given element

Environment: Microsoft Edge version 112.0.1722.68

====

This accessibility issue was found using Accessibility Insights for Web 2.38.1 (axe-core 4.6.3), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
```

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
